src/cryptokit.ml

```diff
@@ -2024,3 +2024,21 @@ let mod_mult a b c =
   Bn.to_bytes ~numbits:(String.length c * 8)
     (Bn.mod_ (Bn.mult (Bn.of_bytes a) (Bn.of_bytes b))
              (Bn.of_bytes c))
+
+let const_time_eq eq len a b =
+  if len a = len b
+  then
+    let l = len a in
+    let lst = List.init l (fun x -> x) in
+    let fold acc i =
+      acc && (eq a b i) in
```

Comment:
If there is a difference in the middle of the string, then for bytes after that point, `eq` will no longer be called (because `&&` is a "short-circuiting" operator). This will be a small difference, but may still be detectable with high resolution timing attacks or cache-line shenanigans.